### PR TITLE
Tweak book

### PR DESCRIPTION
**Pull request by Schaeff**
I went through the entire book fixing mistakes or making improvements.

Goes with https://github.com/powdr-labs/powdr-hello-world/pull/1

### CODE REVIEW

File: book/src/installation.md [hunk 1]
```diff
@@ -44,18 +43,6 @@ With SIMD support for the provers that support it:
 RUSTFLAGS='-C target-cpu=native' cargo install --git https://github.com/powdr-labs/powdr --features halo2,plonky3,simd powdr-cli
 ```
 
-Or, by manually building from a local copy of the [powdr repository](https://github.com/powdr-labs/powdr):
```
Comment (leonardoalt) [inline book/src/installation.md:47]: why remove?

File: pipeline/tests/asm.rs [hunk 2]
```diff
@@ -487,7 +487,7 @@ mod book {
 #[should_panic = "Witness generation failed."]
 fn hello_world_asm_fail() {
     let f = "asm/book/hello_world.asm";
-    let i = [1];
+    let i = [2];
```
Comment (leonardoalt) [inline pipeline/tests/asm.rs:490]: why?

File: powdr-test/examples/hello_world.rs [hunk 3]
```diff
@@ -5,7 +5,7 @@ fn main() {
 
     halo2_pipeline(
         "test_data/asm/book/hello_world.asm",
-        vec![0.into()],
+        vec![1.into()],
```
Comment (leonardoalt) [inline powdr-test/examples/hello_world.rs:8]: why

**Review by leonardoalt — COMMENTED** [review]
looks good, left a few comments about the test

File: book/src/pil/builtins.md [hunk 4]
```diff
@@ -160,7 +160,7 @@ let std::prover::max_degree: -> int
 let std::prover::degree: -> int
 ```
 
-Returns the number of rows / the length of the witness columns, also
+`degree` returns the number of rows / the length of the witness columns, also
```
Comment (chriseth) [inline book/src/pil/builtins.md:163]:
```suggestion
The function `degree` returns the number of rows / the length of the witness columns, also
```

File: powdr-test/examples/hello_world.rs (hunk 3 above)
Comment (Schaeff) [reply powdr-test/examples/hello_world.rs:8]: reverted

File: book/src/SUMMARY.md [hunk 5]
```diff
@@ -37,6 +38,7 @@
     - [Valida](./frontends/valida.md)
     - [EVM](./frontends/evm.md)
 - [Backends](./backends/README.md)
+    - [plonky3](./backends/plonky3.md)
```
Comment (leonardoalt) [inline book/src/SUMMARY.md:41]: is this file missing?

Comment (Schaeff) [reply book/src/SUMMARY.md:41]: no, it was just never referenced :(

Comment (leonardoalt) [reply book/src/SUMMARY.md:41]: ah I see

**Review by leonardoalt — APPROVED** [review]

**Review by leonardoalt — APPROVED** [review]

**Pull request merged**